exporter/oci/export.go

```diff
@@ -77,6 +77,14 @@ func (e *imageExporter) Resolve(ctx context.Context, opt map[string]string) (exp
 		return nil, err
 	}
 
+	if e.opt.Variant == VariantDocker {
```

Comment:
I'm not sure if this is really correct. This is just mediatypes. Users can export with `type=oci` and export fine to Docker as well. But I don't have better ideas.

We may need to check in `buildx` that "docker is multi-platform capable" or "direct push to registry" and only then add "inline-only".

Reply:
Buildx is a possibility, agreed. Not sure though, I have the feeling that that could provide for really tricky results for if we support multiple exporters (#2760).

Could we take the approach from https://github.com/moby/buildkit/pull/3389#issuecomment-1346522020 and never use inline attestations to change from exporting from a manifest to an index? Though, for simple cases, that means we wouldn't get the provenance.